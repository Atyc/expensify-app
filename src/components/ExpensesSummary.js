import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import getExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpensesSummary = (props) => (
    <div>
        Viewing {props.expenses.length}
        {props.expenses.length === 1 ? ' expense ' : ' expenses '}
        totalling {numeral(getExpensesTotal(props.expenses) / 100).format('$0,0.00')}
    </div>
)

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    }
}

export default connect(mapStateToProps)(ExpensesSummary);