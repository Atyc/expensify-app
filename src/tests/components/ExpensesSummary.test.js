import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render expense summary component correctly with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={23} expensesTotal={123456789} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render expense summary component correctly with no expenses', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={0} expensesTotal={0} />);
    expect(wrapper).toMatchSnapshot();
});
