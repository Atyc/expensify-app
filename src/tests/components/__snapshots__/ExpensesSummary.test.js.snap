// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense summary component correctly with multiple expenses 1`] = `
<div>
  Viewing 
  23
   
  expenses
   totalling 
  $1,234,567.89
</div>
`;

exports[`should render expense summary component correctly with no expenses 1`] = `
<div>
  Viewing 
  0
   
  expenses
   totalling 
  $0.00
</div>
`;
